Prod 682 author image fix

AndrejRadomiljac opened the pull request:
✅ Add Linear ID (e.g., `PROD-123`) to PR title to automatically link issue

- Description
- Link to the task https://linear.app/gator/issue/PROD-682/add-deck-author-to-deck-info-page. 
- Admin should be able to add author name in "Deck Author"section while creating or editing deck.
- Admin should be able to add author image in "Deck Author Image" section while creating or editing deck.
- Both filed are optional
- Validation for image is set to the same "Default image size" that we have through whole app
- Deck author validation is set to minimum 5 chars. We can add upper limit later
- You'll see on many places border colour update, I've updated it to the colours that should be there. But the update was required due to Avatar image default border colour that was overwriting any colour from the outside.
- What are the steps to test that this code is working?
- Migrations for deck table with two new columns
- Screen shots or recordings for UI changes
End results:
- Deck preview state with author image and author name
<img width="776" alt="Screenshot 2025-01-23 at 16 45 48" src="https://github.com/user-attachments/assets/87db6069-5b36-4e5c-a8c2-85457279a332" />
- Deck preview state with author name, but no author image
<img width="776" alt="Screenshot 2025-01-23 at 16 46 26" src="https://github.com/user-attachments/assets/cac7a3f0-be83-420b-823b-a68b76d9ccb0" />
- Deck preview state with author image, but no name
<img width="776" alt="Screenshot 2025-01-23 at 16 48 53" src="https://github.com/user-attachments/assets/20a9f628-c2d3-4c9e-8f64-8c91a559598d" />
- Deck preview state with no author image and no name
<img width="776" alt="Screenshot 2025-01-23 at 16 50 33" src="https://github.com/user-attachments/assets/d42251d6-bead-4999-9058-dbe337affeba" />

For not yet active decks we do not show author data
<img width="776" alt="Screenshot 2025-01-23 at 16 51 18" src="https://github.com/user-attachments/assets/d2676235-f062-4711-b26c-be3326ff9423" />

AndrejRadomiljac commented:
Here is the link to preview https://chomp-p8qbs1zfn-gator-labs.vercel.app/login?next=%2Fapplication%3FvercelToolbarCode%3Dmyij3T06BdyowH2. Add your account as admin, create deck using author image and name, make sure that your deck is active and you are good to go.

marvinmarnold commented on app/schemas/deck.ts:
```diff
@@ -47,6 +62,26 @@ export const deckSchema = z
         },
       ),
     description: z.string().min(5).optional().nullish().or(z.literal("")),
+    author: z.string().min(5).optional().nullish().or(z.literal("")),
```
Can we change the min to 2 chars?

AndrejRadomiljac replied:
Sure 👍

marvinmarnold reviewed (changes requested):
The preview image looks a lot smaller for me than your screenshots

https://chomp-dev-git-prod-682-author-image-fix-gator-labs.vercel.app/application/decks/809

![Screenshot 2025-01-27 at 10 16 13](https://github.com/user-attachments/assets/cf1a0139-116c-4131-8445-cd5c6f1257c9)

marvinmarnold reviewed (changes requested):
Incorrect error message. Question description is in fact too short. But author name is optional. But the error message makes it seem like author name is required. Although the error message is misleading, actual behavior is correct. author name is actually treated as optional.
![Screenshot 2025-01-27 at 10 17 23](https://github.com/user-attachments/assets/0d8dbcd4-249f-4fa9-b3bf-9e4159f2d5e2)

AndrejRadomiljac commented:
> The preview image looks a lot smaller for me than your screenshots
> 
> https://chomp-dev-git-prod-682-author-image-fix-gator-labs.vercel.app/application/decks/809
> 
> ![Screenshot 2025-01-27 at 10 16 13](https://private-user-images.githubusercontent.com/1311030/406997958-cf1a0139-116c-4131-8445-cd5c6f1257c9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzc5OTUxMDgsIm5iZiI6MTczNzk5NDgwOCwicGF0aCI6Ii8xMzExMDMwLzQwNjk5Nzk1OC1jZjFhMDEzOS0xMTZjLTQxMzEtODQ0NS1jZDVjNmYxMjU3YzkucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDEyNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAxMjdUMTYyMDA4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZmM4NjAyMDhhMDk4NTdkNzYxM2YxMmUxNDZkM2Y2Y2Q1ZThmMDBiNTQ3YWJlMTFkODkwMDRjZmI1YWY5YzExNCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.e2LePHTW2kMnovzgctioLU8cKF6IR4qC-ROqK8d5teU)

I'll check but it could be due image (effect). It should be 24x24

AndrejRadomiljac commented:
@marvinmarnold I've fixed validation comments. Author name is now 2 chars instead of 5. Messages are fixed on all places including description, heading etc. One thing I've added, so if you type one char for author name and fill all other required parts, on submit will block deck creation since you've not achieved valid char limit even if it is optional. If you remove one char and leave that spot empty you will be able to submit successfully.

Regarding image size. Your image size is in proper dimension 24x24. The only thing that is tricky here is purple background and combination of colours of your image.
![Screenshot 2025-01-27 at 22 08 32](https://github.com/user-attachments/assets/ea9102bf-69e2-4212-a0f3-25a3669b53ef)

marvinmarnold commented:
> @marvinmarnold I've fixed validation comments. Author name is now 2 chars instead of 5. Messages are fixed on all places including description, heading etc. One thing I've added, so if you type one char for author name and fill all other required parts, on submit will block deck creation since you've not achieved valid char limit even if it is optional. If you remove one char and leave that spot empty you will be able to submit successfully.

Great

> Regarding image size. Your image size is in proper dimension 24x24. 

Oh, turns out I was confused because I thought the deck image in your screeshots was the author image. So I was wondering why your images were so much bigger. But I was just looking at the wrong thing.
